### PR TITLE
PW-3167 State Data validator

### PR DESCRIPTION
**Pull request by maassenbas**
## Summary
Deserialise the StateData into the models browserInfo, riskData, paymentMethod
This will validate the keys of the state data. 

## Tested scenarios

**Fixed issue**:

### CODE REVIEW

File: force-app/main/default/staticresources/adyenCheckout.js [hunk 1]
```diff
@@ -19,7 +19,9 @@ function renderAdyenComponent(paymentMethodsResponse) {
 }
 
 function handleOnChange(state){
-    //handleOnChange
+   if(state.isValid){
+        document.getElementById('adyenStateData').value = JSON.stringify(state.data);
```
[the remark this answers is not in the document]
Comment (maassenbas) [reply force-app/main/default/staticresources/adyenCheckout.js:23]: @cyattilakiss frontend validation will be handled in ticket PW-3064, which I'm currently implementing.

File: force-app/main/default/classes/AdyenPaymentRequest.cls [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            return paymentRequest;
+       }
+
+       public static PaymentMethodDetails getPaymentMethodDetails(Map<String, Object> paymentMethod){
+            String paymentType = (String)paymentMethod.get('type');
+            System.Debug('#### paymentType = ' + paymentType);
+            if(paymentType == 'scheme'){
+                CardDetails cardDetails = new CardDetails();
+                cardDetails.holderName = (String)paymentMethod.get('holderName');
+                cardDetails.encryptedCardNumber = (String)paymentMethod.get('encryptedCardNumber');
```
[the remark this answers is not in the document]
Comment (maassenbas) [reply force-app/main/default/classes/AdyenPaymentRequest.cls:31]: I will look into this now

[the remark this answers is not in the document]
Comment (maassenbas) [reply force-app/main/default/classes/AdyenPaymentRequest.cls:31]: Apparently not as easy to create a loop here and assign these values to the model. 
There is no default support for this in Apex. 
I might be able to write a helper class where I serialise it, get/set the value and deserialise it again.
But I think that is not the best solution for it for now.

File: force-app/main/default/classes/AdyenPaymentRequest.cls [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                Map<String, Object> paymentMethod = (Map<String, Object>) stateDataObjects.get('paymentMethod');
+                paymentRequest.paymentMethod = getPaymentMethodDetails(paymentMethod);
+            }
+
+            return paymentRequest;
+       }
+
+       public static PaymentMethodDetails getPaymentMethodDetails(Map<String, Object> paymentMethod){
+            String paymentType = (String)paymentMethod.get('type');
+            System.Debug('#### paymentType = ' + paymentType);
```
Comment (msilvagarcia) [inline force-app/main/default/classes/AdyenPaymentRequest.cls:27]: You can delete this one :)

**Review by msilvagarcia — CHANGES_REQUESTED** [review]

**Review by msilvagarcia — APPROVED** [review]

**Review by cyattilakiss — APPROVED** [review]

**Pull request merged**